### PR TITLE
Improve index check for compatibility with different versions of the DDOS 

### PR DESCRIPTION
**Pull request by acl**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/discovery/storage/datadomain.inc.php [hunk 1]
```diff
@@ -22,6 +22,9 @@
         $total = $storage['fileSystemSpaceSize'] * $units;
         $used = $storage['fileSystemSpaceUsed'] * $units;
         if ($descr == "/data: post-comp") {
+            if (is_null($index)) {
+                $index=$tmp_index;
```
Comment (murrant) [inline includes/discovery/storage/datadomain.inc.php]: $tmp_index is undefined...

Comment (murrant) [reply includes/discovery/storage/datadomain.inc.php]: Why use fileSystemResourceIndex at all?

Comment (acl) [reply includes/discovery/storage/datadomain.inc.php]: I was torn on this as well. Older DataDomain versions do have the fileSystemResourceIndex on their MIB file where as modern DataDomains have completely removed it. 

I don't know if it served any purpose at all in the past. I agree with your change and I'll keep testing to see if it's used in different setups. Thanks.

Comment (murrant) [reply includes/discovery/storage/datadomain.inc.php]: someone probably just didn't realize fileSystemResourceIndex  is in the table index.

**Pull request merged**